### PR TITLE
 [4.0] refactored .Math stylecop errors

### PR DESCRIPTION
**Pull request by jvbsl**
### Purpose of this PR

* stylecop.props import readded to .Math and fixed all stylecop errors
* fixed some xml comment inconsistencies
* fixed missing file documentation at top of file(in .Core/Utility/BlittableValueType{T}.cs)

### Testing status
Well it does build and it shouldn't have any major implementation changes.

### Comments
fixed some stuff of https://github.com/opentk/opentk/pull/802 and now in a feature branch. I hope the simple naming is enough.
I did not reorder static variables, because there still seems to be a discussion about that.

### CODE REVIEW

File: src/OpenTK.Mathematics/Matrix/Matrix3x2.cs [hunk 1]
```diff
@@ -71,15 +72,11 @@ public Matrix3x2(Vector2 row0, Vector2 row1, Vector2 row2)
...
-        (
-            float m00,
-            float m01,
-            float m10,
-            float m11,
-            float m20,
-            float m21
-        )
+        [SuppressMessage("ReSharper", "SA1117", Justification = "For better readability of Matrix struct.")]
+        public Matrix3x2(
```
Comment (FreezyLemon) [inline src/OpenTK.Mathematics/Matrix/Matrix3x2.cs]: I like the idea of these constructor "shapes", but OpenTK uses the "Allman" brace style, which means you should move the parentheses like this:
```csharp
public Matrix2x3
(
    // arguments
    // more arguments
)
```

File: src/OpenTK.Mathematics/Data/Quaternion.cs [hunk 2]
```diff
@@ -191,7 +191,7 @@ public Vector4 ToAxisAngle()
         /// <summary>
         /// Returns a copy of the Quaternion scaled to unit length.
         /// </summary>
-        /// <returns>The normalized copy.</returns>
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Data/Quaternion.cs]: Possible loss of information. Should probably keep as "copy" instead of "Quaternion". We know it's a quaternion.

File: src/OpenTK.Mathematics/Matrix/Matrix2.cs [hunk 3]
```diff
@@ -367,19 +366,19 @@ public static Matrix2 Mult(Matrix2 left, float right)
...
         {
-            float lM11 = left.Row0.X,
-                lM12 = left.Row0.Y,
-                lM21 = left.Row1.X,
-                lM22 = left.Row1.Y,
-                rM11 = right.Row0.X,
-                rM12 = right.Row0.Y,
-                rM21 = right.Row1.X,
-                rM22 = right.Row1.Y;
+            float leftM11 = left.Row0.X,
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Matrix/Matrix2.cs]: Don't declare multiple variables this way. Each one gets its own separate declaration.

File: src/OpenTK.Mathematics/Matrix/Matrix3.cs [hunk 4]
```diff
@@ -781,7 +777,7 @@ public static void Invert(ref Matrix3 mat, out Matrix3 result)
             {
                 { mat.Row0.X, mat.Row0.Y, mat.Row0.Z },
                 { mat.Row1.X, mat.Row1.Y, mat.Row1.Z },
-                { mat.Row2.X, mat.Row2.Y, mat.Row2.Z }
+                { mat.Row2.X, mat.Row2.Y, mat.Row2.Z },
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Matrix/Matrix3.cs]: Do not add trailing commas.

File: src/OpenTK.Mathematics/Matrix/Matrix3x4.cs [hunk 5]
```diff
@@ -706,52 +699,52 @@ public static Matrix3x4 Mult(Matrix3x4 left, Matrix3x4 right)
...
+            result.Row1.X = (leftM21 * rightM11) + (leftM22 * rightM21) + (leftM23 * rightM31);
+            result.Row1.Y = (leftM21 * rightM12) + (leftM22 * rightM22) + (leftM23 * rightM32);
+            result.Row1.Z = (leftM21 * rightM13) + (leftM22 * rightM23) + (leftM23 * rightM33);
+            result.Row1.W = (leftM21 * rightM14) + (leftM22 * rightM24) + (leftM23 * rightM34) + leftM24;
+            result.Row2.X = (leftM31 * rightM11) + (leftM32 * rightM21) + (leftM33 * rightM31);
+            result.Row2.Y = (leftM31 * rightM12) + (leftM32 * rightM22) + (leftM33 * rightM32);
+            result.Row2.Z = (leftM31 * rightM13) + (leftM32 * rightM23) + (leftM33 * rightM33);
+            result.Row2.W = (leftM31 * rightM14) + (leftM32 * rightM24) + (leftM33 * rightM34) + leftM34;
+
+            /*result.Row0 = (right.Row0 * leftM11 + right.Row1 * leftM12 + right.Row2 * leftM13);
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Matrix/Matrix3x4.cs]: If this comment is not used, remove it.

File: src/OpenTK.Mathematics/Matrix/Matrix3x4d.cs [hunk 6]
```diff
@@ -312,6 +304,7 @@ public void Invert()
         /// <param name="axis">The axis to rotate about.</param>
         /// <param name="angle">Angle in radians to rotate counter-clockwise (looking in the direction of the given axis).</param>
         /// <param name="result">A matrix instance.</param>
+        [SuppressMessage("Microsoft.StyleCop.CSharp.DocumentationRules", "SA1305", Justification = "Is not hungarian notation but abbreveation for precalculated values.")]
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Matrix/Matrix3x4d.cs]: What error is this actually suppressing?

File: src/OpenTK.Mathematics/Data/Quaternion.cs (hunk 2 above)
Comment (jvbsl) [reply src/OpenTK.Mathematics/Data/Quaternion.cs]: well I'm not too sure on that I mean it's a struct, of course it's a copy

Comment (Nihlus) [reply src/OpenTK.Mathematics/Data/Quaternion.cs]: It's more informative than Quaternion; copy semantics are an important facet of the mathematics subsystem.

File: src/OpenTK.Mathematics/Matrix/Matrix3.cs [hunk 7]
```diff
@@ -354,7 +349,7 @@ public Matrix3 Inverted()
         /// <summary>
         /// Returns a copy of this Matrix3 without scale.
         /// </summary>
-        /// <returns>The matrix.</returns>
+        /// <returns>The matrix without scaling.</returns>
```
Comment (jvbsl) [inline src/OpenTK.Mathematics/Matrix/Matrix3.cs]: What aboout these ones, because The return value comment here does not specify that it is a copy. Perhaps we should mark such functions as "Pure" as well?

Comment (Nihlus) [reply src/OpenTK.Mathematics/Matrix/Matrix3.cs]: Marking them as pure is a good idea. In general, JetBrains annotations should be used to annotate contracts, purity, and nullability.

Comment (jvbsl) [reply src/OpenTK.Mathematics/Matrix/Matrix3.cs]: I left this be at first. And I think I will do pure functions in a separate commit(perhaps even separate PR?).
Question is do we want to have the annotations in the public API as well? Because Jetbrains recommends that if we to, to include it directly into the SourceCode instead of using the nuget package.

Comment (Nihlus) [reply src/OpenTK.Mathematics/Matrix/Matrix3.cs]: We'll be using Fody to generate external annotations, which is what we're already doing in the other projects.

**Comment by jvbsl** [comment]
Could you review again please? Don't know if you saw that I did a new commit with hopefully all changes ;)

File: src/OpenTK.Mathematics/Data/Quaternion.cs [hunk 8]
```diff
@@ -210,7 +210,7 @@ public void Invert()
         /// <summary>
         /// Returns a copy of this Quaterniond with its rotation angle reversed.
         /// </summary>
-        /// <returns>The inverted copy.</returns>
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Data/Quaternion.cs]: Same thing here, loss of information.

File: src/OpenTK.Mathematics/Vector/Vector4h.cs [hunk 9]
```diff
@@ -129,7 +129,7 @@ public Vector4h(float x, float y, float z, float w, bool throwOnError)
         /// <summary>
         /// Initializes a new instance of the <see cref="Vector4h"/> struct.
         /// </summary>
-        /// <param name="v">The <see cref="Vector4"/> to convert.</param>
```
Comment (Nihlus) [inline src/OpenTK.Mathematics/Vector/Vector4h.cs]: Why this change? This constructor explicitly converts a normal vector.

**Comment by Nihlus** [comment]
Nice job! Just a few nitpicks left.

File: src/OpenTK.Mathematics/Vector/Vector4h.cs (hunk 9 above)
Comment (jvbsl) [reply src/OpenTK.Mathematics/Vector/Vector4h.cs]: good quetion. I can't remember to have done this probably happened while merging...

File: src/OpenTK.Mathematics/MathHelper.cs [hunk 10]
```diff
@@ -189,23 +189,20 @@ public static float InverseSqrtFast(float x)
         /// which is found in the Quake III source code. This implementation comes from
         /// http://www.codemaestro.com/reviews/review00000105.html. For the history of this method, see
         /// http://www.beyond3d.com/content/articles/8/.
+        /// double magic number from: https://cs.uwaterloo.ca/~m32rober/rsqrt.pdf
+        /// chapter 4.8.
         /// </remarks>
         public static double InverseSqrtFast(double x)
```
Comment (jvbsl) [inline src/OpenTK.Mathematics/MathHelper.cs]: fixed this one as well, because it was a pretty easy and fast one. And I think it would still fit in here, because everything was in .Math.

**Comment by Nihlus** [comment]
Spankin'. Thanks a ton!
